apps/autoscale-by-metric.html.markerb

```diff
@@ -0,0 +1,181 @@
+
+```cmd
+fly secrets set -o my-autoscaler --stage FAS_API_TOKEN="FlyV1 ..."
+```
+
+## Acquire a token to read from Prometheus
+
+The next auth token you'll need is one that has permissions to read from your organization's Prometheus data on Fly:
+
+```cmd
+fly tokens create readonly -o my-org
+```
+
+Copy the token and use it as a secret on your autoscaler app:
+
+```cmd
+fly secrets set -o my-autoscaler --stage FAS_PROMETHEUS_TOKEN="FlyV1 ..."
+```
+
+### Configure your autoscaler fly.toml
+
+Next, set up a `fly.toml` configuration file for your autoscaler to set a few environment variables. Replace the `my-autoscaler`, `my-target-app`, and `my-org` with values for your situation.
+
+```toml
+app = "my-autoscaler"
+
+[build]
+image = "flyio/fly-autoscaler:0.3"
+
+[env]
+FAS_PROMETHEUS_ADDRESS = "https://api.fly.io/prometheus/my-org"
+FAS_PROMETHEUS_METRIC_NAME = "qdepth"
+FAS_PROMETHEUS_QUERY = "sum(queue_depth{app='$APP_NAME'})"
+
+FAS_APP_NAME = "my-target-app"
+FAS_CREATED_MACHINE_COUNT = "min(50, qdepth / 2)"
+
+[metrics]
+port = 9090
+path = "/metrics"
+```
+
+Environment variables are the primary way to define your configuration.
+
+Metrics collection settings:
+
+- `FAS_PROMETHEUS_ADDRESS` defines the Prometheus URL endpoint to query.
+- `FAS_PROMETHEUS_METRIC_NAME` defines the local variable name that the metric result will be store as. This example stores the query result value as `qdepth`.
+- `FAS_PROMETHEUS_QUERY` defines the Prometheus query to run. This example computes the sum of a user-defined `queue_depth` metric.
+
+Autoscaling settings:
+
+- `FAS_APP_NAME` is the name of the target application to scale.
+- `FAS_CREATED_MACHINE_COUNT` defines an [Expr](https://expr-lang.org/) expression to calculate the number of Machines to create.
+
+This example expression assumes that each Machine could handle two items in the queue and uses the `min()` function to prevent the autoscaler from scaling more than `50` Machines.
+
+### Deploy the autoscaler
+
+The autoscaler only works on a single Machine, so you'll use the `--ha` option to turn off the high availability feature that cerates two Machines:
+
+```cmd
+fly deploy --ha=false
+```
+
+After the autoscaler deploys, you should see the number of Machines in your target application increase as your user-defined `queue_depth` gauge increases.
+
+<div class="note icon">
+**Note:** The autoscaler creates new Machines in an application by cloning existing ones so it is not able to scale to zero. It will always keep at least one Machine running.
+</div>
+
+You can find a full working example of the autoscaler at our [fly-autoscaler-example](https://github.com/fly-apps/fly-autoscaler-example) repo.
+
+## Other use cases
+
+### Start and stop Machines
+
+If you already have a pool of created Machines that you wish to autoscale, you can use the `FAS_STARTED_MACHINE_COUNT` expression instead of creating and destroying Machines.
```

Comment:
@benbjohnson Would this be better?
```suggestion
If you already have a pool of created Machines that you want to autoscale, you can use the `FAS_STARTED_MACHINE_COUNT` expression instead of creating and destroying Machines with `FAS_CREATED_MACHINE_COUNT`.
```